src/yukon/backend/src/api/controllers/nodes/__init__.py

```diff
@@ -1,129 +1,256 @@
+            body = await request.get_json()
+            return body['value'], 200
+
+        @self.nodes_controller.route('/<int:nodeId>', methods=['GET'])
+        async def node_details(nodeId) -> Tuple[Response, None]:
+            class NodeGetDetailsResponse(object):
+                def __init__(self, name: str, id: int, health: str, mode: str,
+                             uptime: int, vendor_code: int, software_version: str,
+                             crc: str, hardware_version: str, uid: str, authenticity: str) -> None:
+                    self.name = name
+                    self.id = id
+                    self.health = health
+                    self.mode = mode
+                    self.uptime = uptime
+                    self.vendor_code = vendor_code
+                    self.software_version = software_version
+                    self.crc = crc
+                    self.hardware_version = hardware_version
+                    self.uid = uid
+                    self.authenticity = authenticity
+
+                def serialise(self) -> Dict[str, Any]:
+                    return {
+                        'name': self.name,
+                        'id': self.id,
+                        'health': self.health,
+                        'mode': self.mode,
+                        'uptime': self.uptime,
+                        'vendorCode': self.vendor_code,
+                        'softwareVersion': self.software_version,
+                        'crc': self.crc,
+                        'hardware_version': self.hardware_version,
+                        'uid': self.uid,
+                        'authenticity': self.authenticity
+                    }
+
+            # For now, send a mock reponses as an example
+            mock_responses = NodeGetDetailsResponse('node_0', nodeId, 'OK', 'OPERATIONAL', 200,
+                                                    990, '4.3.2.1', '0xTOOMUCHBEEF', '1.2.3.4', 'my-awesome-uid',
+                                                    'I am authentic')
+
+            return jsonify(mock_responses.serialise())
+
+        @self.nodes_controller.route('/<int:nodeId>/shutdown', methods=['PUT'])
+        async def node_shutdown(nodeId) -> Tuple[Response, int]:
+            return "", 200
+
+        @self.nodes_controller.route('/<int:nodeId>/firmwareupdate', methods=['PUT'])
+        async def node_firmware_update(nodeId, param) -> Tuple[Response, int]:
+            body = await request.get_json()
+            return body['name'], 200
+
+    @property
+    def nodes_controller(self):
+        return self._nodes_controller
+
+    @property
+    def node_list(self):
+        return self._node_list
+
+    @property
+    def nodeid_list(self):
+        return self._nodeid_list
+
+    @classmethod
+    def health_to_text(self, health_code):
+        node_health_code = {uavcan.node.Heartbeat_1_0.HEALTH_NOMINAL: 'NOMINAL',
```

Comment:
the mappings in these class methods should be defined as psedo-private (i.e. `_node_health_code`) on the class. You end up creating these maps for every invocation of the class method otherwise.